gammapy/background/energy_offset_array.py

```diff
@@ -0,0 +1,135 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from astropy.units import Quantity, UnitsError
+from astropy.coordinates import Angle
+from astropy.table import Table
+from astropy.units import Quantity
+from astropy.coordinates import Angle
+
+__all__ = [
+    'EnergyOffsetArray',
+]
+
+
+class EnergyOffsetArray(object):
+    """
+    Energy offset dependent array.
+
+    Parameters
+    ----------
+    energy : `~astropy.units.Quantity`
+         energy bin vector
+    offset : `~astropy.coordinates.Angle`
+        offset bin vector
+    data : `~numpy.ndarray`
+        data array (2D)
+
+    """
+
+    def __init__(self, energy=None, offset=None, data=None):
+        self.energy = Quantity(energy, 'TeV')
+        self.offset = Angle(offset, 'deg')
+        self.data = data
+
+    def fill_events(self, event_list):
+        """
+        from the fill_events method of the Cubebackgroundmodel class in model.py
+        Fill events in a 2D energy_offset histograms
+
+        Get data files corresponding to the observation list, histogram
+        the counts and fill the corresponding cube containers.
+
+        Parameters
+        -------------
+        events : list of `~gammapy.data.EventList`
+           Python list of event list objects.
+            
+        
+        """
+
+        self.data = None
+        # loop over the Lost of object EventList
+        for (i, data_set) in enumerate(event_list):
+            ev_cube_hist = self.fill_one_event_list(data_set)
+            # fill data
+            if (i == 0):
+                self.data = ev_cube_hist
+            else:
+                self.data += ev_cube_hist
+
+                
+    def fill_one_event_list(self, events):
+        """
+        histogram the counts of an EventList object in 2D (energy,offset)
+
+        Parameters
+        -------------
+        events :`~gammapy.data.EventList`
+           Event list objects.
+                   
+        """
+        
+        try:
+            ev_detx = Angle(events['DETX'])
+            ev_dety = Angle(events['DETY'])
+            ev_energy = Quantity(events['ENERGY'])
+        except UnitsError:
+            ev_detx = Angle(events['DETX'], 'deg')
+            ev_dety = Angle(events['DETY'], 'deg')
+            ev_energy = Quantity(events['ENERGY'],
+                                 events.meta['EUNIT'])
+        # ici calculer offset mis voir si je le calcul avec detx dety ou raddec
+        offset = np.sqrt(ev_detx ** 2 + ev_dety ** 2)
```

Comment:
This offset calculation should be split off into a property or method on the `EventList` class, so that here you can simply do:
```
offset = events.offset
```
(event offset will be needed in other places as well).